File: Medium/413. Arithmetic Slices/Solution.java
class Solution {
    public int numberOfArithmeticSlices(int[] A) {
        if (A == null || A.length < 3)
            return 0;

        List<Integer> sequences = new ArrayList<>();

        int trioles = 0;
        for (int i = 0; i < A.length - 2; i++) {
            if (A[i + 2] - A[i + 1] == A[i + 1] - A[i])
                trioles++;
            else if (trioles > 0) {
                sequences.add(2 + trioles);
                trioles = 0;
            }
        }

        if (trioles > 0)
            sequences.add(2 + trioles);

        int count = 0;
        for (int s : sequences) {
            System.out.println(s);
            count += slices(s);
        }

        return count;
    }

    int slices(int n) {
        if (n < 3)
            return 0;
        if (n == 3)
            return 1;

        return slices(n - 1) + (n - 2);
    }
}

File: Medium/413. Arithmetic Slices/solution.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var numberOfArithmeticSlices = function (nums) {
  let slices = 0;
  let dp = new Array(nums.length).fill(0);

  for (let i = 2; i < nums.length; i++) {
    if (nums[i - 2] - nums[i - 1] == nums[i - 1] - nums[i]) {
      dp[i] = 1 + dp[i - 1];
      slices += dp[i];
    }
  }

  return slices;
};
